src/Riok.Mapperly/Descriptors/MappingBuilders/EnumerableMappingBuilder.cs

```diff
@@ -200,13 +201,7 @@ or CollectionType.IEnumerable
         if (existingMapping != null)
             return new DelegateMapping(ctx.Source, ctx.Target, existingMapping);
 
-        return new ForEachAddEnumerableMapping(
-            sourceType,
-            targetType,
-            elementMapping,
-            AddMethodName,
-            ctx.CollectionInfos!.Source.CountPropertyName
-        );
+        return new ForEachAddEnumerableMapping(sourceType, targetType, elementMapping, AddMethodName, CountPropertyName);
```

Comment:
I think your indentation is correct, I think the problem also exists in `EnumerableMappingBuilder.BuildEnumerableToArrayMapping`.
Also currently we accept `Length` as count property names of custom collections, don't we? I think a custom collection type as source which only implements `IEnumerable<T>` with a `int Length` property won't work with this fix.

Another approach would probably be in 
`EnumerableMappingBuilder.BuildCollectionTypeForICollection` instead of only returning the `INamedTypeSymbol`, also return the name of the count property:
```c#
private static (ITypeSymbol Type, string? CountPropertyName) BuildCollectionTypeForICollection(MappingBuilderContext ctx, CollectionInfo info)
    {
        return info.ImplementedTypes.HasFlag(CollectionType.IReadOnlyCollection)
            ? (BuildCollectionType(ctx, CollectionType.IReadOnlyCollection, info.EnumeratedType), nameof(IReadOnlyCollection<object>.Count))
            : info.ImplementedTypes.HasFlag(CollectionType.ICollection)
                ? (BuildCollectionType(ctx, CollectionType.ICollection, info.EnumeratedType), nameof(ICollection<object>.Count))
                : (info.Type, info.CountPropertyName);
    }
```

or alternatively we could probably return an entire new `CollectionInfo` for `IReadOnlyCollection<T>` / `ICollection<T>` (I'd probably prefer this 🤔).